files/build_templates/buffers_config.j2

```diff
@@ -128,6 +128,16 @@ def
 {%- endfor %}
 {%- set port_names_active  = port_names_list_active  | join(',') %}
 
+{%- set port_names_list_extra_queues = [] %}
+{%- for port in PORT_ACTIVE %}
+    {% if (generate_queue_buffers_with_extra_lossless_queues is defined or generate_queue_buffers_with_extra_lossless_queues_with_inactive_ports is defined) and 
+            ('type' in DEVICE_METADATA['localhost'] and DEVICE_METADATA['localhost']['type'] == 'LeafRouter' and DEVICE_NEIGHBOR_METADATA is defined and DEVICE_NEIGHBOR[port].name in DEVICE_NEIGHBOR_METADATA and DEVICE_NEIGHBOR_METADATA[DEVICE_NEIGHBOR[port].name].type == 'ToRRouter') or
+            ('subtype' in DEVICE_METADATA['localhost'] and DEVICE_METADATA['localhost']['subtype'] == 'DualToR' and DEVICE_NEIGHBOR_METADATA is defined and DEVICE_NEIGHBOR[port].name in DEVICE_NEIGHBOR_METADATA and DEVICE_NEIGHBOR_METADATA[DEVICE_NEIGHBOR[port].name].type == 'LeafRouter') %}
```

Comment:
can you put these in brackets since they should be evaluated first before the 'and' 
macro present and (type is leaf or dualtor)

Reply:
Fixed.